service-base/src/main/java/org/eclipse/hono/service/NewAbstractApplication.java

```diff
@@ -46,7 +46,7 @@
      */
     protected final Logger log = LoggerFactory.getLogger(getClass());
 
-    private final Set<ObjectFactory<? extends AbstractServiceBase>> serviceFactories = new HashSet<>();
+    private Set<ObjectFactory<? extends AbstractServiceBase>> serviceFactories;
```

Comment:
IMHO we should keep the initialization in order to keep our own, private copy of the object factories.